config/config.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/onflow/flow-go/config/network"
+)
+
+var (
+	conf = viper.New()
+
+	//go:embed config.yml
+	configFile string
```

Comment:
Right now the yml file essentially acts as an encoded representation of config defaults. Do we want to allow node operators to actually provide a `config.yml` file to configure their node as well? I think doing so would be beneficial (and mostly leverage functionality already provided by Viper). 

But if we support that, we'll need to adjust this package to accept a file input from the node scaffold, and rename `configFile` etc. to make it clear it is only defaults.

Reply:
https://github.com/onflow/flow-go/pull/4340/commits/c353c042a9acd9e5fb0cc315401470789f64dd2a